tests/NATS.Client.KeyValueStore.Tests/KeyValueContextTest.cs

```diff
@@ -0,0 +1,122 @@
+
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(60));
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        var store = await kv.CreateStoreAsync(buketName, cancellationToken);
+        var status = await store.GetStatusAsync(cancellationToken);
+        status.Info.Config.Description.Should().BeNull();
+
+        var natsKVConfig = new NatsKVConfig(buketName) { Description = expectedDescription };
+        await kv.UpdateStoreAsync(natsKVConfig, cancellationToken);
+        var updatedStatus = await store.GetStatusAsync(cancellationToken);
+        updatedStatus.Info.Config.Description.Should().Be(natsKVConfig.Description);
+    }
+
+    [Fact]
+    public async Task Delete_store_test()
+    {
+        var bucketName = "kv1";
+
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        await kv.CreateStoreAsync(bucketName, cancellationToken);
+
+        var result = await kv.DeleteStoreAsync(bucketName, cancellationToken);
+
+        result.Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task Get_bucket_names_test()
+    {
+        var actualBucketNames = new List<string>();
+        var expectedBucketNames = new List<string> { "kv1", "kv2", "kv3" };
+
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        foreach (var bucketName in expectedBucketNames)
+        {
+            await kv.CreateStoreAsync(new NatsKVConfig(bucketName), cancellationToken);
+        }
+
+        await foreach (var bucketName in kv.GetBucketNamesAsync(cancellationToken))
+        {
+            actualBucketNames.Add(bucketName);
+        }
+
+        actualBucketNames.Should().BeEquivalentTo(expectedBucketNames);
+    }
+
+    [Fact]
+    public async Task Get_statuses_test()
+    {
+        var bucketName = "kv1";
+        var expectedBucketName = $"KV_{bucketName}";
+
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+        var cancellationToken = cts.Token;
+
+        await using var server = NatsServer.StartJS();
+        await using var nats = server.CreateClientConnection();
+
+        var js = new NatsJSContext(nats);
+        var kv = new NatsKVContext(js);
+
+        var natsKVConfig = new NatsKVConfig(bucketName) { Compression = true };
```

Comment:
Compression isn't supported in server 2.9.x which we're still testing against so it's failing the build. maybe use some other property e.g. max_msgs here.

Reply:
Fixed. Strange, now fallen another `Reconnect_doesnt_drop_partially_sent_msgs `

Reply:
don't worry. that's a flapper. I'll re-run it.